src/shogun/kernel/Kernel.h

```diff
@@ -235,16 +235,14 @@ class CKernel : public CSGObject
 			REQUIRE(rhs, "CKernel::get_kernel_diagonal(): Right-handside "
 						"features missing!\n");
 
-			REQUIRE(lhs->get_num_vectors()==rhs->get_num_vectors(),
```

Comment:
the old version do not consider the asymmetric case of a kernel matrix such as `kernel->init(features, inducing_features)`